permission-element.bs

```diff
@@ -19,7 +18,408 @@ Abstract: A `<permission>` HTML element to request browser permissions in-page.
+
+* <dfn attribute for="HTMLPermissionElement">\[[Types]]</dfn> is null
+    or an [=ordered set=] of [=powerful features=]. Null represents the
+    uninitialized state, which allows the value to be modified. The empty
+    list &laquo;[]&raquo; is the state in which no permission applies, and which
+    will no longer allow modification. Note that the
+    {{HTMLPermissionElement/type}} property reflects this internal state.
+
+## <{permission}> element interesting behaviours ## {#permission-element-very-interesting}
+
+### The {{HTMLPermissionElement/type}} property ### {#permission-element-type-property}
+
+The <{permission}> element has a few surprising behaviours, to support its
+security properties:
+
+The permission type cannot be modified. Modifying the permission type at will
+may lead to user confusion, and hence we'd like to prevent it. Since, however,
+a page may create a <{permission}> element dynamically we still need to offer
+an API to modify it. To do do, we distinguish between a freshly initialized and
+an empty or invalid (no permission) state, where the former allows setting the
+type and the latter does not.
+
+Example:
+```js
+// Changing a valid type:
+var pepc = document.createElement("permission");
+pepc.type = "camera";  // Okay.
+pepc.type;  // "camera".
+pepc.type = "geolocation";  // Not okay. Would have been okay as initial assignment.
+pepc.type;  // "camera". Reflects the internal state, which has not changed.
+
+// Setting an invalid type:
+pepc = document.createElement("permission");
+pepc.type = "icecream";  // Ice cream is not a powerful browser feature. Not okay.
+pepc.type;  // "". Reflects the internal state.
+pepc.type = "camera";  // Still Not okay, because type as already been set.
+                       // Would have been okay as initial assignment.
+pepc.type;  // "". Reflects the internal state, which has not changed.
+
+```
+
+<div algorithm="HTMLPermissionElement/types getter">
+The HTMLPermissionElement's {{HTMLPermissionElement/type}} getter steps are:
+
+1. If {{[[Types]]}} is null: Return `""`.
+1. Return a string, containing the concatenation of all [=powerful feature=]
+    names in {{[[Types]]}}, seperated by " ".
+
+</div>
+
+<div algorithm="HTMLPermissionElement/types setter">
+The HTMLPermissionElement's {{HTMLPermissionElement/type}} setter steps are:
+
+1. If {{[[Types]]}} is not null: Return.
+1. Set {{[[Types]]}} to &laquo;[]&raquo;.
+1. Parse the input as a string of [=powerful feature=] names, seperated by whitespace.
+1. If any errors occured, return.
+1. Check if the set of [=powerful features=] is supported for the {{HTMLPermissionElement}} by the user agent. If not, return.
+1. Append each [=powerful feature=] name to the {{[[Types]]}} [=ordered set=].
+
+</div>
+
+### Activation blockers ### {#permission-element-activation-blockers}
+
+<pre class=idl>
+enum PermissionElementInvalidReason {
+  "",  // No blocker reason.
+  "style", "type_count", "illegal_subframe", "covered",
```

Comment:
If this is based on the explainer it's out-of-date, I would suggest using the list below for our up-to-date list (and I'll take an item to update the explainer).

Reply:
Done.